spec/mailers/provider_mailer_spec.rb

```diff
@@ -5,6 +5,17 @@
 
   subject(:mailer) { described_class }
 
+  let(:course_option) { course_option_for_provider_code(provider_code: 'ABC') }
+  let(:application_choice) do
+    build_stubbed(:submitted_application_choice,
+                  course_option: course_option,
+                  application_form:
+                    build_stubbed(
+                      :completed_application_form,
+                   ))
+  end
+  let(:provider_user) { application_choice.provider.provider_users.first }
```

Comment:
This goes against the conventions we've adopted elsewhere to avoid `let`. I think this example is pretty clear with the `let`s personally. @tijmenb pointed me to https://thoughtbot.com/blog/lets-not (you've probably already seen it!).

Reply:
One way to refactor it without using the `let`s would be to have a plain method that set up the test fixtures with args as needed for any variations in those fixtures.

Reply:
I know, we have plenty of existing unit tests with let blocks, I thought this is more of an issue with the system tests, let me know what do you think @tijmenb